File: C++/majorityelement.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution
{
  public:
   int majorityelement(vector<int>&nums)
    {
      
      unordered_map<int,int>hash;
      for(size_t i=0;i<nums.size();i++)
      {
         if(nums.size()/2 < ++hash[nums[i]])
          return nums[i];

     }
 }
  };

int main()
{
  Solution s;
  vector<int>input;
  input.emplace_back(2);
  input.emplace_back(2);
  input.emplace_back(2);
  input.emplace_back(1);
  input.emplace_back(1);
  input.emplace_back(1);
  input.emplace_back(2);
  input.emplace_back(2);
  cout<<"Majority_element:"<<s.majorityelement(input)<<endl;

  return 0;
}

File: Python/array/majority_element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        dict_count = {}
        for i in range(0,len(nums)):
            if nums[i] in dict_count:
                dict_count[nums[i]] += 1
            else:
                dict_count[nums[i]] = 1
        for (key,values) in dict_count.items():
            if values >= len(nums)/2:
                return key
